app/models/document.rb

```diff
@@ -230,10 +230,7 @@ def publish
 
   def unpublish
     handle_remote_error do
-      Services.publishing_api.unpublish(content_id, type: 'gone')
-
-      AttachmentDeleteWorker.perform_async(content_id)
-      RummagerDeleteWorker.perform_async(base_path)
+      DocumentUnpublisher.unpublish(content_id, base_path)
```

Comment:
Despite the seeming redundancy with the name of the receiver, I'm in favour of methods with names that describe what they do.  It's not clear what a method called `call` actually does without reading the method, whereas a method called `unpublish` gives us a hint.